### PR TITLE
Enhancing GET, POST and PUT request support

### PR DESCRIPTION
**Pull request by Abhinandangatti**
Adding Support for GET, POST, POST requests with basic and custom authorization.

### CODE REVIEW

File: pkg/utils/http-utils.go [hunk 1]
```diff
@@ -76,6 +76,76 @@ func GetHeadersCustomRetry(URL string, headers map[string]string, retryCount int
...
+	return GetHeadersCustomAuthCustomRetry(URL, headers, -1, -1, token)
+}
+
+// GetHeadersCustomAuthRetry makes an HTTP call to given URL with headers and custom auth
+// and returns response. It also retries for failures.
+func GetHeadersCustomAuthRetry(URL string, headers map[string]string, token string) (Response, error) {
+	return GetHeadersCustomAuthCustomRetry(URL, headers, defaultRetryCount, defaultRetryWaitSeconds, token)
+}
+
+// Get makes an HTTP call to given URL with headers and custom auth
```
Comment (vishweshwarp) [inline pkg/utils/http-utils.go]: Update the comment to match func name

File: pkg/utils/http-utils.go [hunk 2]
```diff
@@ -129,6 +199,147 @@ func PostHeadersFormDataCustomRetry(URL string, formData map[string]string, head
...
+			SetBody(requestBody)
+	}
+
+	resp, err := request.
+		SetHeaders(headers).
+		Put(URL)
+	return processResponse(resp), err
+}
+
+// Put makes an HTTP call to given URL with basic authentication
```
Comment (vishweshwarp) [inline pkg/utils/http-utils.go]: same here and all the below funcs

**Review by vishweshwarp — COMMENTED** [review]
please fix the comments

File: pkg/utils/http-utils.go (hunk 1 above)
Comment (Abhinandangatti) [reply pkg/utils/http-utils.go]: Done updated the comments

File: pkg/utils/http-utils.go (hunk 2 above)
Comment (Abhinandangatti) [reply pkg/utils/http-utils.go]: Done updated the comments

**Review by vishweshwarp — APPROVED** [review]

**Pull request merged**